regression/check_submodules.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+filter='.[]'
+filter+='|select(.name == \"$path\")'
+filter+='|select(.name == \"$name\")'
+filter+='|select(.commit == \"$sha1\")'
+filter="\"$filter\""
+
+git submodule foreach --quiet "jq -e $filter ../wit-manifest.json" || { \
```

Comment:
Since it's not completely obvious what this line is doing, could you write a short comment describing what you are doing here? This file itself is just called `check_submodules.sh`, but it's not exactly clear what we are checking the submodules for. And even though I know you are checking the git submodules against the wit manifest, the algorithm you are using to check for correctness is not readily apparent without having to dig into exactly what the `git submodule foreach` command and the `jq` filter is doing.

My current understanding is that this script loops over each git submodule, and for each submodule it attempts to locate a record (JSON object) within `wit-manifest.json` which simultaneously matches the name, path, and commit hash of the git submodule.

Note that this will miss cases where the wit manifest specifies a repository that is not present as a git submodule. I am not sure if that is intentional.

Reply:
intentional. since the wit-manifest may have extra packages like `api-chisel3-sifive` that shouldn't be submodules

Reply:
OK, if it's intentional, then let's capture that intent in a comment, since it's not possible to infer that from just the code alone.